qt/Session.cc

```diff
@@ -1133,6 +1143,32 @@ void Session::addTorrent(AddData const& add_me, tr_variant* args, bool trash_ori
     q->run();
 }
 
+void Session::onDuplicatesTimer()
+{
+    decltype(duplicates_) duplicates;
+    duplicates.swap(duplicates_);
+
+    QStringList lines;
+    for (auto it : duplicates_)
+    {
+        lines.push_back(tr("%1 (copy of %2)")
+            .arg(it.first)
+            .arg(it.second.left(7)));
+    }
+
+    if (!lines.empty())
+    {
+        lines.sort(Qt::CaseInsensitive);
+        auto* d = new QMessageBox(QMessageBox::Warning,
+            tr("Unable to add Duplicate Torrent(s)", "", lines.size()),
+            lines.join(QStringLiteral("\n")),
```

Comment:
I don't think stuffing a huge list into the `text` property would look nice (does it show a scroll bar at least?). `QMessageBox` has a [`detailedText` property](https://doc.qt.io/qt-5/qmessagebox.html#detailedText-prop); if you don't feel like it (I do though), at least moving the list to [`informativeText` property](https://doc.qt.io/qt-5/qmessagebox.html#informativeText-prop) would look nicer on some platforms.

![](https://doc.qt.io/qt-5/images/msgbox4.png)

I guess we could e.g. include first name in the text, and put the rest in the details if there're more than one?

Reply:
Fixed in 26b5ce5